tests/commonTemplates_test.go

```diff
@@ -355,6 +330,131 @@ var _ = Describe("Common templates", func() {
+			UpdateFunc: func(t *templatev1.Template) {
+				t.Parameters = nil
+			},
+			EqualsFunc: func(old *templatev1.Template, new *templatev1.Template) bool {
+				return reflect.DeepEqual(old.Parameters, new.Parameters)
+			},
+		}
+
+		testTemplateS390x = testResource{
+			Name:           templateName + "-" + string(architecture.S390X),
+			Namespace:      strategy.GetTemplatesNamespace(),
+			Resource:       &templatev1.Template{},
+			ExpectedLabels: expectedLabels,
+			UpdateFunc: func(t *templatev1.Template) {
+				t.Parameters = nil
+			},
+			EqualsFunc: func(old *templatev1.Template, new *templatev1.Template) bool {
+				return reflect.DeepEqual(old.Parameters, new.Parameters)
+			},
+		}
+
+		updateSsp(func(foundSsp *ssp.SSP) {
+			foundSsp.Spec.EnableMultipleArchitectures = ptr.To(true)
+			foundSsp.Spec.Cluster = &ssp.Cluster{
+				WorkloadArchitectures: []string{
+					string(architecture.AMD64),
+					string(architecture.ARM64),
+					string(architecture.S390X),
+				},
+				ControlPlaneArchitectures: []string{nodeArchitecture},
+			}
+		})
+		waitUntilDeployed()
+	})
+
+	AfterAll(func() {
+		strategy.RevertToOriginalSspCr()
+	})
+
+	DescribeTable("created resource", func(res *testResource, arch architecture.Arch) {
+		found := res.NewResource()
+		Expect(apiClient.Get(ctx, res.GetKey(), found)).To(Succeed())
+		Expect(found.GetLabels()).To(HaveKeyWithValue(commonTemplates.TemplateArchitectureLabel, string(arch)))
+	},
+		Entry("[test_id:TODO] template for amd64", decorators.Conformance, &testTemplateAmd64, architecture.AMD64),
+		Entry("[test_id:TODO] template for arm64", decorators.Conformance, &testTemplateArm64, architecture.ARM64),
+		Entry("[test_id:TODO] template for s390x", decorators.Conformance, &testTemplateS390x, architecture.S390X),
+	)
+
+	DescribeTable("should restore modified resource", expectRestoreAfterUpdate,
+		Entry("[test_id:TODO] template for amd64", decorators.Conformance, &testTemplateAmd64),
+		Entry("[test_id:TODO] template for arm64", decorators.Conformance, &testTemplateArm64),
+		Entry("[test_id:TODO] template for s390x", decorators.Conformance, &testTemplateS390x),
+	)
+
+	DescribeTable("recreate after delete", expectRecreateAfterDelete,
+		Entry("[test_id:TODO] template for amd64", decorators.Conformance, &testTemplateAmd64),
+		Entry("[test_id:TODO] template for arm64", decorators.Conformance, &testTemplateArm64),
+		Entry("[test_id:TODO] template for s390x", decorators.Conformance, &testTemplateS390x),
+	)
+
+	Context("when disabled multi-arch", Ordered, func() {
+		BeforeAll(func() {
+			updateSsp(func(foundSsp *ssp.SSP) {
+				foundSsp.Spec.EnableMultipleArchitectures = ptr.To(false)
+			})
+			waitUntilDeployed()
+		})
+
+		AfterAll(func() {
+			updateSsp(func(foundSsp *ssp.SSP) {
+				foundSsp.Spec.EnableMultipleArchitectures = ptr.To(true)
+			})
+			waitUntilDeployed()
+		})
+
+		DescribeTable("should delete template from non-default arch", func(res *testResource, arch architecture.Arch) {
```

Comment:
The expected outcome for the test cases is unclear. What should happen on which arch?